web/site/js/vis/PixiVisualizer.js

```diff
@@ -211,42 +224,123 @@ export default class PixiVisualizer extends VObject {
+            case OtDeviceRole.OT_DEVICE_ROLE_ROUTER:
+                return "Router";
+            case OtDeviceRole.OT_DEVICE_ROLE_LEADER:
+                return "Leader";
+        }
+    }
+
+    modeToString(mode) {
+        let s = "";
+        if (mode.getRxOnWhenIdle()) {
+            s += "r";
+        }
+        if (mode.getSecureDataRequests()) {
+            s += "s";
+        }
+        if (mode.getFullThreadDevice()) {
+            s += "d";
+        }
+        if (mode.getFullNetworkData()) {
+            s += "n";
+        }
+        return s;
+    }
+
     visAddNode(nodeId, x, y, radioRange) {
         let node = new Node(nodeId, x, y, radioRange);
         this.nodes[nodeId] = node;
         this._nodesStage.addChild(node._root);
-        this.setSelectedNode(nodeId)
+        this.setSelectedNode(nodeId);
+
+        let msg = `Added at (${x},${y})`
+        if (!this.real) {
+            msg += `, radio range ${radioRange}`
+        }
+        this.logNode(nodeId, msg)
     }
 
     visSetNodeRloc16(nodeId, rloc16) {
-        this.nodes[nodeId].setRloc16(rloc16)
+        let node = this.nodes[nodeId];
+        let oldRloc16 = node.rloc16;
+        node.setRloc16(rloc16);
+        if (oldRloc16 != rloc16) {
+            this.logNode(nodeId, `RLOC16 changed from ${this.formatRloc16(oldRloc16)} to ${this.formatRloc16(rloc16)}`)
+        }
     }
 
     visSetNodeRole(nodeId, role) {
-        this.nodes[nodeId].setRole(role)
+        let oldRole = this.nodes[nodeId].role;
+        this.nodes[nodeId].setRole(role);
+        if (oldRole != role) {
+            this.logNode(nodeId, `role changed from ${this.roleToString(oldRole)} to ${this.roleToString(role)}`)
+        }
     }
 
     visSetNodeMode(nodeId, mode) {
-        this.nodes[nodeId].setMode(mode)
+        let oldMode = this.nodes[nodeId].nodeMode;
+        this.nodes[nodeId].setMode(mode);
+        this.logNode(nodeId, `mode changed from ${this.modeToString(oldMode)} to ${this.modeToString(mode)}`);
     }
 
     visSetNetworkInfo(version, commit, real) {
+        let oldVersion = this.otVersion;
+        let oldCommit = this.otCommit;
         this.setOTVersion(version, commit);
-        this.setReal(real)
+        this.setReal(real);
+
+        if (oldVersion != this.otVersion) {
+            this.log(`OpenThread Version: ${version}`);
+        }
+        if (oldCommit != this.otCommit) {
+            this.log(`OpenThread Commit: ${commit}`);
+        }
     }
 
     visDeleteNode(nodeId) {
         let node = this.nodes[nodeId];
         delete this.nodes[nodeId];
         node.destroy();
         if (nodeId === this._selectedNodeId) {
-            this.setSelectedNode(0)
+            this.setSelectedNode(0);
         }
+        this.logNode(nodeId, "Deleted")
     }
 
     visSetSpeed(speed) {
         this.speed = speed;
-        this.actionBar.setSpeed(speed)
+        this.actionBar.setSpeed(speed);
+        this.log(`Speed set to ${Math.round(speed)}`)
```

Comment:
Can we also do `if (oldSpeed != speed)`?

Reply:
Checked